dask/base.py

```diff
@@ -80,6 +81,61 @@ def annotate(**annotations):
     ...     B = A + 1
     """
 
+    # Sanity check annotations used in place of
+    # legacy distributed Client.{submit, persist, compute} keywords
+    if "workers" in annotations:
+        if isinstance(annotations["workers"], (list, set, tuple)):
+            annotations["workers"] = list(annotations["workers"])
+        elif isinstance(annotations["workers"], str):
+            annotations["workers"] = [annotations["workers"]]
+        elif callable(annotations["workers"]):
+            pass
+        else:
+            raise TypeError(
+                "'workers' annotation must be a sequence of str or a str. Got %s"
+                % type(annotations["workers"])
+            )
+
+    if (
+        "priority" in annotations
+        and not isinstance(annotations["priority"], Number)
+        and not callable(annotations["priority"])
+    ):
+        raise TypeError(
+            "'priority' annotation must be a Number. Got %s"
```

Comment:
Maybe include callable in the message here

Reply:
done.